frontend/src/redux/auth/actions.ts

```diff
@@ -0,0 +1,34 @@
+import { Dispatch } from "redux"
+
+import { Auth } from "../../types"
+import { UpdateAuthAction, UPDATE_AUTH } from "./action-types"
+
+
+export function updateAuth(auth: Auth): UpdateAuthAction {
+    return {
+        type: UPDATE_AUTH,
+        payload: auth
+    }
+}
+
+// dummy function stubbing out a login call for now
+async function loginApiCall(email: string, password: string): Promise<Auth> {
+  return {
+    is_refreshing: false,
+    auth_token: email,
+    refresh_token: password,
+    last_refresh: new Date(0)
+  }
+}
+
+export function logIn(email: string, password: string) {
+  return async (dispatch: Dispatch): Promise<Auth> => {
+      try {
+          const authResponse = await loginApiCall(email, password)
+          dispatch<UpdateAuthAction>(updateAuth(authResponse))
```

Comment:
I feel like explicitly providing the generic type at the call site is an unusual pattern (it should be inferred based on the arguments)--am I wrong? Does TypeScript require this here?

Reply:
It doesn't require this, and technically w/ intellisense, it is actually being inferred. I can remove this though. I tend to lean in the direction of being overly explicit, but this may not be a place where we need it.